### PR TITLE
Add json schemas

### LINKED ISSUE #36: Add json schemas for object types
#### Proposed solution
JSON-based schemas for object types can be used for following features:
- generating forms to frontend
- validating json data when it is pushed to mongo
- returning schemas to user from /objects/<schema> endpoint #16 

Schemas could be generated from xml files converted to json using backend, so they reflect current status of the backend. Another way is to generate json schemas directly from xsd schemas, but this probably means we have to improve xml-to-json conversion process in backend. 

#### DoD (Definition of Done)
- Each object type has json schema available either on repo or in database.

#### Testing
- no need to test automatically

### PR DESCRIPTION
**Pull request by blankdots**
### Description

Add JSON schemas 

### Related issues

Fixes #36 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Changes Made

1. added JSON schemas for experiment, study, sample, analysis, run, dac, policy, dataset
2. moved to having 2 Schema loaders one for XML and one for JSON
3. added `/schemas/{schema}` endpoint for JSON schema retrieval
4. added tests for newly added functions
5. added a `platform` edge case in `parser.py`
6. renamed `schemas.json` to `ena_schemas.json`

### Testing

- [x] Unit Tests

### Mentions

tested the JSON schemas how they look in UI and agains what we generate (well at least for the examples we had) and here are some thoughts
1. all schemas share attributes and links
   - we can remove linkx `xrefLink` `urlLink` and `entrezLink` don't think we will ever use them
   - we can also keep them as is and when we look at the forms remove what is not needed.
2. in UI we need a mechanism to identify and cross-reference objects
3. `alias`, `centerName` and `identifiers` seemed like metadata that is why I did not add them to the schema
4. for cross-reference we need an `accessionID` or that is what is now, but we can also generate a temp accession ID or an alias
5. there will be `ui:schemas` so @otahontas PR pending.

### CODE REVIEW

File: metadata_backend/helpers/schema_loader.py [hunk 1]
```diff
@@ -32,16 +33,44 @@ def get_schema(self, schema_type: str) -> XMLSchema:
...
+
+
+class JSONSchemaLoader:
+    """JSON Loader implementation."""
+
+    def __init__(self) -> None:
+        """Load schemas folder on initialization."""
+        self.path = SCHEMAS_ROOT
+
+    def get_schema(self, schema_type: str) -> dict:
```
[the remark this answers is not in the document]
Comment (blankdots) [reply metadata_backend/helpers/schema_loader.py:61]: yes, I was lazy on that :) will do.

File: metadata_backend/helpers/schemas/ena_dac.json [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            "title": "Contacts", 
+            "items": {
+                "$ref": "#/definitions/contact"
+            },
+            "contains": {
+                "properties": {
+                  "mainContact": {"const": true}
+                }
+              },
+              "maxContains": 1
```
[the remark this answers is not in the document]
Comment (blankdots) [reply metadata_backend/helpers/schemas/ena_dac.json:128]: i think it is ignored, I will still keep it because we can use in the back-end validation.

File: metadata_backend/helpers/schema_loader.py [hunk 3]
```diff
@@ -16,32 +19,84 @@ class SchemaNotFoundException(Exception):
...
+        """Find schema which is used to match JSON files against.
+
+        :param schema_type: Schema type to be searched for
+        :returns: JSONSchema able to validate XML against defined schema type
+        :raises SchemaNotFoundException: If searched schema doesn't exist
+        """
+        file = self._identify_file(schema_type)
+        with file.open() as f:
+            schema_contet = json.load(f)
```
Comment (otahontas) [inline metadata_backend/helpers/schema_loader.py:101]: typo: `schema_content`

File: metadata_backend/helpers/schemas/ena_experiment.json [hunk 4]
```diff
@@ -0,0 +1,657 @@
...
+            "title": "Experiment Description",
+            "description": "Free-form text describing the data sets.",
+            "type": "string"
+        },
+        "studyRef": {
+            "title": "Study Identifier (Accession ID or title)",
+            "description": "Identifies the associated study.",
+            "$ref": "#/definitions/reference"
+        },
+        "desgin": {
```
Comment (otahontas) [inline metadata_backend/helpers/schemas/ena_experiment.json:552]: typo: design

File: metadata_backend/helpers/schema_loader.py (hunk 3 above)
Comment (blankdots) [reply metadata_backend/helpers/schema_loader.py]: done

File: metadata_backend/helpers/schemas/ena_experiment.json (hunk 4 above)
Comment (blankdots) [reply metadata_backend/helpers/schemas/ena_experiment.json]: done.
